cmd/tetra/debug/progs.go

```diff
@@ -0,0 +1,400 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package debug
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"sync"
+	"text/tabwriter"
+	"time"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/btf"
+	"github.com/cilium/ebpf/link"
+	"github.com/spf13/cobra"
+	"golang.org/x/sys/unix"
+)
+
+type prog struct {
+	id    uint32
+	name  string
+	pin   string
+	cnt   uint64
+	time  time.Duration
+	alive bool
+}
+
+type overhead struct {
+	p    *prog
+	pct  float64
+	cnt  uint64
+	time time.Duration
+}
+
+type progsConfig struct {
+	all     bool
+	lib     string
+	bpffs   string
+	once    bool
+	noclr   bool
+	timeout int
+}
+
+var (
+	initOnce sync.Once
+	initErr  error
+	initProg *ebpf.Program
+	cfg      progsConfig
+)
+
+func NewProgsCmd() *cobra.Command {
+	cmd := cobra.Command{
+		Use:     "progs",
+		Aliases: []string{"top"},
+		Short:   "Retrieve information about BPF programs on the host",
+		Long: `Retrieve information about BPF programs on the host.
+
+Examples:
+- tetragon BPF programs top style
+  # tetra debug progs
+- all BPF programs top style
+  # tetra debug progs --all
+- one shot mode (displays one interval data)
+  # tetra debug progs --once
+- change interval to 10 seconds
+  # tetra debug progs  --timeout 10
+- change interval to 10 seconds in one shot mode
+  # tetra debug progs --once --timeout 10
+`,
+
+		Run: func(_ *cobra.Command, _ []string) {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			cfg.timeout = int(time.Second) * cfg.timeout
```

Comment:
It's converting to nanoseconds, right? I don't like changing unit of an existing field, but it's fine here, just I would add a comment clarifying what's going on maybe.